zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/utils/SearchClientAdapter.java

```diff
@@ -0,0 +1,165 @@
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch.generic.Requests;
+import org.opensearch.client.opensearch.indices.IndexState;
+
+public class SearchClientAdapter {
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final JsonpMapper ELS_JSON_MAPPER =
+      new co.elastic.clients.json.jackson.JacksonJsonpMapper(MAPPER);
+  private static final org.opensearch.client.json.JsonpMapper OPENSEARCH_JSON_MAPPER =
+      new JacksonJsonpMapper(MAPPER);
+
+  private final ElasticsearchClient elsClient;
+  private final OpenSearchClient osClient;
+
+  public SearchClientAdapter(final ElasticsearchClient elsClient) {
+    this.elsClient = elsClient;
+    osClient = null;
+  }
+
+  public SearchClientAdapter(final OpenSearchClient osClient) {
+    elsClient = null;
+    this.osClient = osClient;
+  }
+
+  private JsonNode opensearchIndexToNode(final IndexState index) throws IOException {
+    final var indexAsMap =
+        SchemaResourceSerializer.serialize(
+            JacksonJsonpGenerator::new, (gen) -> index.serialize(gen, OPENSEARCH_JSON_MAPPER));
+
+    return MAPPER.valueToTree(indexAsMap);
+  }
+
+  private JsonNode elsIndexToNode(final co.elastic.clients.elasticsearch.indices.IndexState index)
+      throws IOException {
+    final var indexAsMap =
+        SchemaResourceSerializer.serialize(
+            co.elastic.clients.json.jackson.JacksonJsonpGenerator::new,
+            (gen) -> index.serialize(gen, ELS_JSON_MAPPER));
+
+    return MAPPER.valueToTree(indexAsMap);
+  }
+
+  private JsonNode elsIndexTemplateToNode(final IndexTemplateItem indexTemplate)
+      throws IOException {
+    final var templateAsMap =
+        SchemaResourceSerializer.serialize(
+            co.elastic.clients.json.jackson.JacksonJsonpGenerator::new,
+            (gen) -> indexTemplate.serialize(gen, ELS_JSON_MAPPER));
+
+    return MAPPER.valueToTree(templateAsMap);
+  }
+
+  private JsonNode opensearchIndexTemplateToNode(
+      final org.opensearch.client.opensearch.indices.get_index_template.IndexTemplateItem
+          indexTemplate)
+      throws IOException {
+    final var templateAsMap =
+        SchemaResourceSerializer.serialize(
+            JacksonJsonpGenerator::new,
+            (gen) -> indexTemplate.serialize(gen, OPENSEARCH_JSON_MAPPER));
+
+    return MAPPER.valueToTree(templateAsMap);
+  }
+
+  private JsonNode elsPolicyToNode(final Lifecycle lifecyclePolicy) throws IOException {
+    final var policyAsMap =
+        SchemaResourceSerializer.serialize(
+            co.elastic.clients.json.jackson.JacksonJsonpGenerator::new,
+            (gen) -> lifecyclePolicy.serialize(gen, ELS_JSON_MAPPER));
+
+    return MAPPER.valueToTree(policyAsMap);
+  }
+
+  public JsonNode getIndexAsNode(final String indexName) throws IOException {
+    switch (elsClient != null ? "elsClient" : "osClient") {
```

Comment:
❌ in this switch and the ones below: are we ever going to the `default` case?
we never check if `osClient` is null, so if `elsClient` is null, we assume that the `osClient` is not. And we will never reach the default even if we have no client instantiated.

I suggest we use a simple `if, else`, as in
```java
    if (elsClient != null) {
      // DO something with elsClient
    } else if (osClient != null) {
      // DO something with osClient
    }
    throw new IllegalStateException("Must instantiate client in SearchClientAdapter");

Reply:
you can also `requireNonNull` for the injected client in both constructors, and remove the `throw new IllegalStateException` from all the methods